File: tests/apps/python/src/application/use_cases/activate.py
from dataclasses import dataclass

from domain.model.user import User
from domain.services.policy import ActivationPolicy, can_activate


@dataclass(frozen=True)
class ActivateUser:
    actor: str = "system"

    def execute(self, user: User) -> dict[str, str]:
        if not ActivationPolicy().allows(user):
            return {"status": "blocked"}
        return {"status": "activated", "id": user.user_id}


def build_activation_command(user: User) -> User:
    return user


def activation_label(user: User) -> str:
    return "allowed" if can_activate(user) else "blocked"


def nullable_label(value: str | None) -> str:
    return value or "missing"

File: tests/apps/typescript/src/application/use_cases/activate.ts
import { User } from '../../domain/model/user';
import { ActivationPolicy, canActivate } from '../../domain/services/policy';

export class ActivateUser {
    execute(user: User): string {
        return new ActivationPolicy().allows(user) ? 'activated' : 'blocked';
    }
}

export function buildActivationCommand(user) {
    return user;
}

export function activationLabel(user) {
    return canActivate(user) ? 'allowed' : 'blocked';
}

export function nullableLabel(value: string | undefined): string {
    return value ?? 'missing';
}

export function optionalLabel(value?: string): string {
    return value ?? 'missing';
}
